developer/src/kmc-analyze/test/osk-character-use-warnings.tests.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Keyman is copyright (C) SIL Global. MIT License.
+ */
+import * as fs from 'fs';
+import { assert } from 'chai';
+import 'mocha';
+import { AnalyzeOskCharacterUse } from '../src/osk-character-use/index.js';
+import { TestCompilerCallbacks } from '@keymanapp/developer-test-helpers';
+import { AnalyzerMessages } from '../src/analyzer-messages.js';
+
+describe('AnalyzeOskCharacterUse warnings', function() {
+    const callbacks = new TestCompilerCallbacks();
+    const TMP_NO_COUNTS = '/tmp/mock-map-no-counts.json';
+    const TMP_WITH_COUNTS = '/tmp/mock-map-with-counts.json';
+
+    before(() => {
+        fs.writeFileSync(TMP_NO_COUNTS, JSON.stringify({
+            map: [{ usages: ['U+1780', 'U+1781'] }]
+        }));
+        fs.writeFileSync(TMP_WITH_COUNTS, JSON.stringify({
+            map: [{ usages: [{ char: 'U+1780', count: 2 }] }]
+        }));
+    });
```

Comment:
We should have these as fixtures in developer/src/kmc-analyze/test/fixtures/ folder, rather than writing to /tmp (which is not cross-platform-safe)